jax/numpy/lax_numpy.py

```diff
@@ -979,6 +979,23 @@ def nan_reduction(a, axis=None, out=None, keepdims=False, **kwargs):
 nansum = _make_nan_reduction(onp.nansum, sum, 0, nan_if_all_nan=False)
 nanprod = _make_nan_reduction(onp.nanprod, prod, 1, nan_if_all_nan=False)
 
+@_wraps(onp.nanmean)
+def nanmean(a, axis=None, dtype=None, out=None, keepdims=False):
+  if out is not None:
+    raise ValueError("nanmean does not support the `out` argument.")
+  # Check and Count all non-NaN values
+  nan_mask = logical_not(isnan(a))
+  normalizer = sum(nan_mask, axis=axis, dtype=int32, keepdims=keepdims)
+  normalizer = lax.convert_element_type(normalizer, dtype)
+  #Perform mean calculation
+  if dtype is None:
```

Comment:
If we're going to test and special case integer and bool values anyway, we can simplify this even more. Why not just  move this test to the top of the function, and delegate to `mean` if the input is bool or integer? These types cannot be `NaN`.